spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractPollingEndpoint.java

```diff
@@ -168,6 +169,9 @@ protected void onInit() {
 					this.taskExecutor = new ErrorHandlingTaskExecutor(this.taskExecutor, this.errorHandler);
 				}
 			}
+			if (this.transactionSynchronizationFactory == null) {
+				this.transactionSynchronizationFactory = new PassthroughTransactionSynchronizationFactory();
+			}
```

Comment:
We only want to take the hit of saving the message in a TL __if the poller is transactional__ - we can easily tell that in the XML parser and via the DSL spec, but not so easily via the `@Poller` annotation; we may need to add a `transactional` property to `@Poller` for that.

On the other hand, any "around" advice that can throw an exception may need access to the message if it throws the exception after the proceed.

So, I don't mind if we always apply it if the advice chain is > 0 length - that's probably the simplest solution that would work in all cases while only taking the extra unneeded hit if someone configures an advice chain.